src/vendorpromo/forms.py

```diff
@@ -53,14 +54,83 @@ class Meta:
             'meta',
             'offer']
 
-
+# To be deprecated
 class PromoCodeForm(forms.ModelForm):
 
     class Meta:
         model = Promo
         fields = ['code']
 
 
+class PromotionalCampaignForm(forms.ModelForm):
+    applies_to = forms.ModelMultipleChoiceField(queryset=get_product_model().objects.all(), required=False)
+    is_percent_off = forms.NullBooleanField(widget=forms.RadioSelect(
+        choices=[
+            (True, "Percent Off"),
+            (False, "Fixed Amount")
+        ]
+    ))
+    discount_value = forms.DecimalField(min_value=0, label=_("Discount Value"), required=False)
+
+    class Meta:
+        model = PromotionalCampaign
+        fields = [
+            "campaign_id",
+            "name",
+            "description",
+            "start_date",
+            "end_date",
+            "max_redemptions",
+        ]
+
+    def __init__(self, *args, **kwargs):
+        site = kwargs.pop('site', None)
+        super().__init__(*args, **kwargs)
+
+        if site:
+            self.fields['applies_to'].queryset = get_product_model().objects.filter(site=site)
+
+        if self.instance.pk:
+            self.fields['applies_to'].initial = self.instance.applies_to.products.all()
+            self.fields['is_percent_off'].initial = (True, "Percent Off") if self.instance.is_percent_off else (False, "Fixed Amount")
+            self.fields['discount_value'].initial = self.instance.applies_to.current_price()
+
+    def clean_discount_value(self):
+        discount_value = self.cleaned_data.get('discount_value', 0)
+
+        if discount_value <= 0:
+            raise forms.ValidationError(_("Number must be greater than 0"))
+        
+        if self.cleaned_data.get('is_percent_off'):
+            if not (0 < discount_value > 100):
```

Comment:
is 100% off allowed? if so, this check should be `if not (0 < discount_value >= 100)`
if not, the error message should be between 1 and 99